Source/ASCollectionView.mm

```diff
@@ -960,30 +960,38 @@ - (CGSize)collectionView:(UICollectionView *)collectionView layout:(UICollection
 {
   ASDisplayNodeAssertMainThread();
   NSIndexPath *indexPath = [NSIndexPath indexPathForItem:0 inSection:section];
-  ASCellNode *cell = [self supplementaryNodeForElementKind:UICollectionElementKindSectionHeader
-                                               atIndexPath:indexPath];
-  if (cell.shouldUseUIKitCell && _asyncDelegateFlags.interop) {
+  ASCollectionElement *element = [_dataController.visibleMap supplementaryElementOfKind:UICollectionElementKindSectionHeader
+                                                                            atIndexPath:indexPath];
+  if (element == nil) {
+    return CGSizeZero;
```

Comment:
> I think the answer will come from...in what case can we get here but have a nil element?

I was assuming that clients that use interop can return an item/supplementary view that is not of type `_ASCollectionViewCell`/`_ASCollectionReusableView`, and so the element can be `nil` in these cases.

It turns out that, there is a race condition in `ASDataController` that can cause an element to be `nil`. We also got crashes reported at Pinterest with similar symptoms. More details here: https://github.com/TextureGroup/Texture/pull/420. Now that we've fixed the root cause, I think the `nil` checks in this PR is less of a concern, although I still want to keep them should code correctness.